### PR TITLE
Infiltration Module

### PR DESCRIPTION
**Pull request by mdahlhausen**
- replace adjust_infiltration_to_lower_pressure with adjust_infiltration_to_new_pressure
- move adjust_infiltration_to_prototype_building_conditions
- move surface_component_infiltration_rate
- move sub_surface_component_infiltration_rate
- add tests

Pull request overview
---------------------

Moves methods into an infiltration module

### Pull Request Author

 - [x] Method changes or additions
 - [x] Added tests for added methods
 - [x] If methods have been deprecated, update rest of code to use the new methods
 - [x] Documented new methods using [yard syntax](https://rubydoc.info/gems/yard/file/docs/GettingStarted.md)
 - [x] Resolved yard documentation errors for new code (ran `bundle exec rake doc`)
 - [x] Resolved rubocop syntax errors for new code (ran `bundle exec rake rubocop`)
 - [x] All new and existing tests passes (note at the time of this PR master CI fails; no new tests fail with this PR)

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: method additions, changes, tests
 - [ ] Check rubocop errors
 - [ ] Check yard doc errors
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If a new feature, test the new feature and try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: lib/openstudio-standards/infiltration/infiltration.rb [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    def self.adjust_infiltration_to_prototype_building_conditions(infiltration_rate, initial_pressure: 75.0)
+      alpha = 0.22 # terrain adjustment factor for an urban environment, unitless
+      uh = 4.47 # wind speed, m/s
+      rho = 1.18 # air density, kg/m^3
+      cs = 0.1617 # positive surface pressure coefficient, unitless
+      n = 0.65 # infiltration coefficient, unitless
+
+      # Calculate the typical pressure - same for all building types
+      final_pressure_pa = 0.5 * cs * rho * uh**2
+      adjusted_infiltration_rate = (1.0 + alpha) * infiltration_rate * (final_pressure_pa / initial_pressure)**n
```
Comment (eringold) [inline lib/openstudio-standards/infiltration/infiltration.rb:38]: why not call `adjust_infiltration_to_new_pressure` with these arguments?

File: lib/openstudio-standards/infiltration/infiltration.rb [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+          infil_rate_cfm_per_ft2 = component_infil_rates_cfm_per_ft2[type]['exterior_wall']
+        when 'Floor'
+          infil_rate_cfm_per_ft2 = component_infil_rates_cfm_per_ft2[type]['floor_over_unconditioned']
+        end
+      when 'Ground'
+        case surface_type
+        when 'Wall'
+          infil_rate_cfm_per_ft2 = component_infil_rates_cfm_per_ft2[type]['below_grade_wall']
+        when 'Floor'
+          infil_rate_cfm_per_ft2 = component_infil_rates_cfm_per_ft2[type]['slab_on_grade']
```
Comment (eringold) [inline lib/openstudio-standards/infiltration/infiltration.rb:96]: Table 1 of https://www.pnnl.gov/main/publications/external/technical_reports/PNNL-18898.pdf says below-grade walls and slab-on-grade floors are 'not used in infiltration calculations'.

File: lib/openstudio-standards/infiltration/infiltration.rb [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+          'skylight' => 0.40
+        },
+        'advanced' => {
+          'opaque_door' => 0.20,
+          'loading_dock_door' => 0.20,
+          'swinging_or_revolving_glass_door' => 1.0,
+          'vestibule' => 1.0,
+          'sliding_glass_door' => 0.20,
+          'window' => 0.20,
+          'skylight' => 0.20
```
Comment (eringold) [inline lib/openstudio-standards/infiltration/infiltration.rb:143]: these are 0.4 in Table 1 of https://www.pnnl.gov/main/publications/external/technical_reports/PNNL-18898.pdf

File: test/modules/infiltration/test_infiltration.rb [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+class TestInfiltration < Minitest::Test
+  def setup
+    @infiltration = OpenstudioStandards::Infiltration
+
+    # load model and set up weather file
+    template = '90.1-2013'
+    climate_zone = 'ASHRAE 169-2013-4A'
+    std = Standard.build(template)
+    @model = std.safe_load_model("#{File.dirname(__FILE__)}/../../../data/geometry/ASHRAESecondarySchool.osm")
```
Comment (eringold) [inline test/modules/infiltration/test_infiltration.rb:11]: wish we didnt have to build a template to load an osm. Could use BTAP::FileIO.safe_load_model. Maybe we should move to a standards module?

**Review by eringold — CHANGES_REQUESTED** [review]

File: lib/openstudio-standards/infiltration/infiltration.rb (hunk 1 above)
Comment (mdahlhausen) [reply lib/openstudio-standards/infiltration/infiltration.rb:38]: I could but this is clearer and shorter.  The alpha bit makes it a bit different, which is why we need a whole separate method for the prototype rates to begin with.

File: lib/openstudio-standards/infiltration/infiltration.rb (hunk 2 above)
Comment (mdahlhausen) [reply lib/openstudio-standards/infiltration/infiltration.rb:96]: I'll remove slabs and below grade walls

File: test/modules/infiltration/test_infiltration.rb (hunk 4 above)
Comment (mdahlhausen) [reply test/modules/infiltration/test_infiltration.rb:11]: yes, I plan to port this to a new module and change the test. I'll change the test once I make that change in another branch.

File: lib/openstudio-standards/infiltration/infiltration.rb (hunk 3 above)
Comment (mdahlhausen) [reply lib/openstudio-standards/infiltration/infiltration.rb:143]: I updated the reference.  Seems like the values in the code came from Table 5.7 of Thornton, Brian A, Rosenberg, Michael I, Richman, Eric E, Wang, Weimin, Xie, YuLong, Zhang, Jian, Cho, Heejin, Mendon, Vrushali V, Athalye, Rahul A, and Liu, Bing. Achieving the 30% Goal: Energy and Cost Savings Analysis of ASHRAE Standard 90.1-2010. United States: N. p., 2011. Web. doi:10.2172/1015277.

**Review by eringold — APPROVED** [review]

**Pull request merged**